### ISSUE TITLE
Add table to replication with different primary key

### ISSUE THREAD

**Issue by mridude23** [bucardo/bucardo#179]
Hi all,

Hit some sort of limitation (maybe) when adding tables. I have a table that does not have a primary key, but it has not one but 2 unique indexes. When adding the table to the replication flow it adds the first unique index for that table (it seems it takes it alphabetically) when in fact I want to add it with the second one.

Changing it in goat table throws the following error:
`Could not find a matching unique constraint that provides those columns`
which honestly is odd as the index is there and it's unique.

Is there a way to change the primary key used in this case?

**Comment by machack666** [comment]
Hmm, in theory this should be possible; these are both unique not null indexes?

**Comment by mridude23** [comment]
Yes, I managed to do it another way meanwhile.
Remove that table from replication, transform an index to primary key (as it should have been from the start) and re-add it. This time it was added with the primary key as expected. 
I think it has something to do with the fact that it was failing with some other errors and I purged the delta tables multiple times making the dbs very out-of-sync. (this is hunch of mine).
